src/loaders/ObjectLoader.js

```diff
@@ -632,11 +679,73 @@ Object.assign( ObjectLoader.prototype, {
 			if ( data.visible !== undefined ) object.visible = data.visible;
 			if ( data.userData !== undefined ) object.userData = data.userData;
 
-			if ( data.children !== undefined ) {
+			/*
+			 * SkinnedMesh creates Bone instances as its children in the constructor
+			 * so skip Bone instance creation here.
+			 */
+			if ( data.type === 'SkinnedMesh' ) {
+
+				var scope = this;
+
+				function traverse ( data ) {
+
+					if ( data.children !== undefined ) {
 
-				for ( var child in data.children ) {
+						for ( var child in data.children ) {
 
-					object.add( this.parseObject( data.children[ child ], geometries, materials ) );
+							if ( data.children[ child ].type !== 'Bone' ) {
+
+								var parent;
+
+								if( data.type === 'SkinnedMesh' ) {
+
+									parent = object;
+
+								} else if ( data.type === 'Bone' ) {
+
+									// should make a method THREE.Skeleton.findBoneByName() ?
+									for ( var i = 0, il = object.skeleton.bones.length; i < il; i ++ ) {
+
+										if ( object.skeleton.bones[ i ].name === data.name ) {
+
+											parent = object.skeleton.bones[ i ];
+											break;
+
+										}
+
+									}
+
+								}
+
+								if ( parent ) {
+
+									parent.add( scope.parseObject( data.children[ child ], geometries, materials, skeletons ) );
+
+								}
+
+							} else {
+
+								traverse( data.children[ child ] );
+
+							}
+
+						}
+
+					}
+
+				}
+
+				traverse( data );
+
+			} else {
```

Comment:
I simplified, would you review the code?

My idea is 
If `bone` has non-bone object as child (ex, left-hand bone has gun mesh object),
serialized `Skeleton` stores the object in `sockets`.
 (the name `sockets` is from UE4 bhouston mentioned)